CareKitStore/CareKitStore/Protocols/CoreData/OCKCoreDataStoreProtocol.swift

```diff
@@ -180,6 +180,8 @@ extension OCKCoreDataStoreProtocol {
             let newVersion = addNewVersion(value)
             newVersion.previous = current
             newVersion.uuid = UUID()
+            current.next = newVersion
```

Comment:
I couldn't find in the code where "next" was being set. This seemed like the appropriate place. `testUpdateTaskVersionPushedToRemote()` also tests to see if next is pushed to the remote